README.md

```diff
@@ -62,15 +79,71 @@ check` before proceeding to debug using `bin/webpack`.
+
+If an environment variable is required during the Docker image build, like for example in the Rails asset compilation process, these should be defined in the Dockerfile in the following format.
+
+`ENV VARIABLE_NAME=Value`
+
+When defining variables in the Dockerfile use dummy values for security reasons since they will be committed to Git and they will ultimately be overriden by docker-compose when you launch a container using the image.
+
+#### <a name="documentation-env-vars-local-dev"></a>Local Development
+
+If an environment variable is required for use in the local development environment it should be declared in the .env file in the following format 
+
+`VARIABLE_NAME=Value`
+
+The [.env.example](./.env.example) file contains the essential environment variables that must exist for local development builds to succeed.
+
+#### <a name="documentation-env-vars-docker-compose"></a>Docker Compose
+
+For docker compose to make the necessary environment variables available in the container at run time they must be declared in the relevant docker-compose.yaml file, of which there are three.
+
+* [docker-compose.yml](./docker-compose.yml) - Variables that are required for local dev and the docker image build phase in azure only should be defined the *environment* section in this file. This will in general only be for the database.
+* [docker-compose.override.yml](./docker-compose.override.yml) - This file is used exclusively for local development and imports variables from the .env file so no further changes are required here.
+* [docker-compose.azure.yml](./docker-compose.azure.yml) - This file is exclusively used in the Azure devops pipeline. Any environment variables required in the Azure build/deployment need to be delcared in the *environment* section of this file. **You should only declare the environment variable here, not its value.** The only exception to this is where we need to map variables due to the use of protected variable names, e.g. Rails SECRET_KEY_BASE.
+
+#### <a name="documentation-env-vars-azure-devops"></a>Azure Hosting (DevOps pipeline)
+
+These steps describe the process for making environment variables available to the the Azure DevOps pipeline.
+
+1. Declare the desired variable in the appropriate "variable group" in the Library section of the Azure DevOps site (https://dfe-ssp.visualstudio.com/Become-A-Teacher/_library?itemType=VariableGroups). All variable groups related to apply are suffixed as such and there is a variable group per deployment environment.
+1. In the [azure-pipelines.yml](./azure-pipelines.yml) file there are several changes to be made:
+   1. For each "make" command script step, add your environment variable to the *env* section in the format `ENV_VAR_NAME: $(var_name)` where **ENV_VAR_NAME** is the environment variable name as it should appear in the docker container and **var_name** is the name of the variable defined in the Azure DevOps variable group.
```

Comment:
Should we use `varName` here and below instead of `var_name`, as that's the convention so far?

Reply:
Agreed. Implemented in most recent commit.